### ISSUE TITLE
Proposal: Add @std/kv

### ISSUE THREAD

**Issue by kitsonk** [denoland/std#6123]
**Is your feature request related to a problem? Please describe.**

While Deno KV is great, there are likely some additional capabilities that wouldn't be integrated into the runtime, but would be generally useful for those working with Deno KV.

**Describe the solution you'd like**

I have been making [kv-toolbox](https://jsr.io/@kitsonk/kv-toolbox) available for over a year and I think there are parts of it that are suitable for becoming part of the Deno standard library.

I propose that `@std/kv` be created, initially with the following areas of capability:

- [kv-toolbox/json](https://jsr.io/@kitsonk/kv-toolbox/doc/json) - a set of utilities and types that are able to serialize and deserialize all of the keys and values that are supported by Deno KV in JSON. This is very useful when having to represent Deno KV data in a browser or transmit Deno KV data "over the wire" while being able to accurately preserve the item.
- [kv-toolbox/size_of](https://jsr.io/@kitsonk/kv-toolbox/doc/size_of) - a function that estimates the byte size of a serialized value in Deno KV. Currently the [documentation](https://docs.deno.com/api/deno/~/Deno.Kv) suggests to use `JSON.stringify()` to estimate the size of values. The problem is that is totally ineffective at estimating complex objects which can be stored in Deno KV but do not serialize properly to JSON (e.g. typed arrays, RegExp, Set, Map, Error, Date, bigint, etc.). While taking the byte length of V8 serialize is more accurate it isn't available in Deploy and is 10x slower than `sizeOf()`.

These capabilities would be fairly straightforward contributions, are fairly well documented and have pretty extensive test coverage.

There is a lot more to `kv-toolbox`, but my opinion is that aren't as generally applicable as the above areas, work around potential future features of Deno KV, or promote patterns that might come with risks, although I would be glad to contribute any part of `kv-toolbox` to Deno.

**Describe alternatives you've considered**

Just continuing to maintain `kv-toolbox`, but limiting the usefulness to a wider audience.

**Comment by kt3k** [comment]
What is the main usage of `sizeOf`? Is it for knowing the size of item before actually writing to KV database to avoid hitting the size limit?

**Comment by kitsonk** [comment]
@kt3k correct, in order to avoid having to attempt a transaction and catch the throw. One solution is to use V8 serialize. But it isn't exposed on Deploy and isn't available in a browser. Outside of usage in Deno, if you were to use `sizeOf()` to estimate a size of a value you wish to serialize in a browser and send to a Deno to store in Deno KV, you would want to try to enforce that limit in the browser before attempting to send it to the server and handle an exception response.

A few other libraries leveraging have run into similar problems (outside of kv-toolbox).

Ref: https://github.com/denoland/deploy_feedback/issues/662
Ref: https://github.com/oliver-oloughlin/kvdex/issues/130
